File: iosApp/iosApp/ComponentViews/RouteIcon.swift
//
//  RouteIcon.swift
//  iosApp
//
//  Created by Brady, Kayla on 5/15/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Foundation
import Shared
import SwiftUI

func routeIcon(_ route: Route) -> Image {
    routeIcon(route.type)
}

func routeIcon(_ routeType: RouteType) -> Image {
    Image(routeIconResource(routeType))
}

func routeIconResource(_ routeType: RouteType) -> ImageResource {
    switch routeType {
    case .bus: .modeBus
    case .commuterRail: .modeCr
    case .ferry: .modeFerry
    default: .modeSubway
    }
}

func routeSlashIcon(_ route: Route) -> Image {
    routeSlashIcon(route.type)
}

func routeSlashIcon(_ routeType: RouteType) -> Image {
    Image(routeSlashIconResource(routeType))
}

func routeSlashIconResource(_ routeType: RouteType) -> ImageResource {
    switch routeType {
    case .bus: .modeBusSlash
    case .commuterRail: .modeCrSlash
    case .ferry: .modeFerrySlash
    default: .modeSubwaySlash
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/RouteIcon.kt
package com.mbta.tid.mbta_app.android.component

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.painter.Painter
import androidx.compose.ui.res.painterResource
import com.mbta.tid.mbta_app.android.R
import com.mbta.tid.mbta_app.model.Route
import com.mbta.tid.mbta_app.model.RouteType

@Composable fun routeIcon(route: Route) = routeIcon(route.type)

@Composable
fun routeIcon(routeType: RouteType) =
    when (routeType) {
        RouteType.BUS -> painterResource(id = R.drawable.mode_bus)
        RouteType.COMMUTER_RAIL -> painterResource(id = R.drawable.mode_cr)
        RouteType.FERRY -> painterResource(id = R.drawable.mode_ferry)
        else -> painterResource(id = R.drawable.mode_subway)
    }

@Composable
fun routeSlashIcon(routeType: RouteType): Painter {
    return when (routeType) {
        RouteType.BUS -> painterResource(id = R.drawable.mode_bus_slash)
        RouteType.COMMUTER_RAIL -> painterResource(id = R.drawable.mode_cr_slash)
        RouteType.FERRY -> painterResource(id = R.drawable.mode_ferry_slash)
        else -> painterResource(id = R.drawable.mode_subway_slash)
    }
}
